server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -674,12 +674,41 @@ private void validateDropColumns(List<DropColumn> dropColumns) {
         }
     }
 
+    /**
+     * Propagates the changes occurred to child columns to the parent columns inner types, e.g.::
+     *   ALTER TABLE ADD COLUMN o['o2']['o3']['x'] INT;
+     *   ==> after adding the column `x`, column `o`, `o['o2']`, and `o['o2']['o3']` also need to append `x` to its inner types.
+     * @param column the column changed
+     * @param type the type that the column changed to (null if dropped)
+     * @param newReferences all references of the current table including the changed references
+     */
+    private void fixParentsInnerTypes(ColumnIdent column,
```

Comment:
```suggestion
    private void updateParentObjectTypes(ColumnIdent column,
```

We so far hardly use the `fix` term for methods